xonsh/completers/tools.py

```diff
@@ -0,0 +1,33 @@
+import builtins
+
+
+def get_filter_function():
+    csc = builtins.__xonsh_env__.get('CASE_SENSITIVE_COMPLETIONS')
+    if csc:
+        def filt(s, x): return s.startswith(x)
+    else:
+        def filt(s, x): return s.lower().startswith(x.lower())
+    return filt
+
+
+def is_iterable(x):
+    try:
+        _ = iter(x)
+        return True
+    except:
+        return False
+
+
+def justify(s, max_length, left_pad=0):
```

Comment:
Yeah, it is worth looking into :).  Anytime we can push code burden into the standard library, we probably should!